test/integration/smoke/test_restore_vm.py

```diff
@@ -148,9 +148,13 @@ def test_03_restore_vm_with_disk_offering_custom_size(self):
         self.assertEqual(root_vol.state, 'Ready', "Volume should be in Ready state")
         self.assertEqual(root_vol.size, 16 * 1024 * 1024 * 1024, "Size of volume and custom disk size should match")
 
-        old_root_vol = Volume.list(self.apiclient, id=old_root_vol.id)[0]
-        self.assertEqual(old_root_vol.state, "Destroy", "Old volume should be in Destroy state")
-        Volume.delete(old_root_vol, self.apiclient)
+        if self.hypervisor.lower() == "vmware":
+            old_root_vol = Volume.list(self.apiclient, id=old_root_vol.id)
+            self.assertEqual(old_root_vol, None, "Old volume should be deleted")
+        else:
+            old_root_vol = Volume.list(self.apiclient, id=old_root_vol.id)[0]
+            self.assertEqual(old_root_vol.state, "Destroy", "Old volume should be in Destroy state")
+            Volume.delete(old_root_vol, self.apiclient)
```

Comment:
@weizhouapache these changes are required for tests with expunge=True in restore operation (update test_02_restore_vm_with_disk_offering and test_01_restore_vm as well)